tests/shanghai/eip4895_withdrawals/test_withdrawals.py

```diff
@@ -31,763 +22,352 @@
-        }
-    elif test_case == ZeroAmountTestCases.FOUR_ONE_WITH_MAX:
-        withdrawals = all_withdrawals
-        post = all_post
-    elif test_case == ZeroAmountTestCases.FOUR_ONE_WITH_MAX_REVERSED:
-        for i, w in enumerate(reversed(all_withdrawals)):
-            withdrawals.append(
-                Withdrawal(
-                    index=i,
-                    validator_index=w.validator_index,
-                    address=w.address,
-                    amount=w.amount,
-                )
-            )
-        post = all_post
-
-    blockchain_test(
-        pre=pre,
-        # TODO: Fix in BlockchainTest? post: Mapping[str, Account | object]
-        # to allow for Account.NONEXISTENT
-        post=post,
-        blocks=[Block(withdrawals=withdrawals)],
-        tag=test_case.value,
-    )
+    # Call to non-existent contract succeeds but does nothing
+    post: Dict[str, Any] = {}
+
+    blockchain_test(pre=pre, post=post, blocks=blocks)
 
 
-@pytest.mark.xdist_group(name="bigmem")
-def test_large_amount(
+def test_withdrawal_zero_amount(
     blockchain_test: BlockchainTestFiller,
     pre: Alloc,
 ) -> None:
     """
-    Test withdrawals that have a large gwei amount.
-
-    Test such that that (gwei * 1e9) could overflow uint64 but not uint256.
+    Test withdrawal with zero amount.
+    Edge case: system call with zero-value withdrawal.
     """
-    withdrawals: List[Withdrawal] = []
-    amounts: List[int] = [
-        (2**35),
-        (2**64) - 1,
-        (2**63) + 1,
-        (2**63),
-        (2**63) - 1,
-    ]
+    pre[DEPOSIT_CONTRACT] = Account(
+        code=get_minimal_deposit_contract_code(),
+        nonce=1,
+    )
 
-    post = {}
-
-    for i, amount in enumerate(amounts):
-        addr = pre.fund_eoa(0)
-        withdrawals.append(
-            Withdrawal(
-                index=i,
-                validator_index=i,
-                address=addr,
-                amount=amount,
-            )
-        )
-        post[addr] = Account(balance=(amount * ONE_GWEI))
+    withdrawal = Withdrawal(
+        index=0,
+        validator_index=0,
+        address=Address(0x01),
+        amount=0,  # Zero amount
+    )
+
+    blocks = [Block(withdrawals=[withdrawal])]
+    post = {DEPOSIT_CONTRACT: Account(storage={})}
 
-    blocks = [
-        Block(
-            withdrawals=withdrawals,
-        )
-    ]
     blockchain_test(pre=pre, post=post, blocks=blocks)
 
 
-@pytest.mark.xdist_group(name="bigmem")
-@pytest.mark.parametrize("amount", [0, 1])
-@pytest.mark.with_all_precompiles
-@pytest.mark.slow()
-def test_withdrawing_to_precompiles(
+def test_withdrawal_max_amount(
     blockchain_test: BlockchainTestFiller,
     pre: Alloc,
-    precompile: int,
-    amount: int,
 ) -> None:
-    """Test withdrawing to all precompiles for a given fork."""
-    sender = pre.fund_eoa()
-    post: Dict = {}
+    """
+    Test withdrawal with maximum uint64 amount.
+    Ensures no overflow in amount handling.
```

Comment:
Same, if you don't record the values called into the contract this test is not very useful